dspMetadataGUI/collectMetadata.py

```diff
@@ -1328,5 +1338,141 @@ def on_key_down(self, evt):
+        box.Add(output_panel, 0, wx.ALL | wx.EXPAND, 7)
+        box.Add(wx.StaticLine(panel, -1), 0, wx.EXPAND | wx.TOP | wx.BOTTOM, 10)
+        # Run Conversion
+        self.btn_convert = wx.Button(panel, label="Convert")
+        self.btn_convert.Bind(wx.EVT_BUTTON, lambda x: self._run_conversion())
+        box.Add(self.btn_convert, 0, wx.ALL | wx.EXPAND, 7)
+        panel.SetSizer(box)
+        self._refresh()
+
+    def _run_conversion(self):
+        """Button `Convert` has been clicked."""
+        if self.__in_files:
+            print(f'saving to: {self.__out_dir}')
+            res = converter.convert_and_save(self.__in_files, self.__out_dir)
+            print(f"Converted: {res} files")
+            open_file(self.__out_dir)
+
+    def _on_select_output(self, label: wx.StaticText):
+        """Select output/target directory by opening a DirDialog."""
+        with wx.DirDialog(self, "Select Output Folder") as dirDialog:
+            res = dirDialog.ShowModal()
+            if res == wx.ID_OK:
+                self.__out_dir = dirDialog.GetPath()
+                label.SetLabel(self.__out_dir)
+            else:
+                self.__out_dir = None
+                label.SetLabel("No Output Directory Selected")
+        self._refresh()
+
+    def _on_select_input(self, label: wx.StaticText, mode):
+        """Select input file(s) by opening FileDialog/DirDialog, depending on the selected mode."""
+        if mode == ConverterDialog.InType.SINGLE_FILE:
+            with wx.FileDialog(self, "Select Input File", wildcard="*.ttl") as fileDialog:
+                res = fileDialog.ShowModal()
+                if res == wx.ID_OK:
+                    path = fileDialog.GetPath()
+                    self.__in_files = [path]
+                    label.SetLabel(str(self.__in_files))
+        elif mode == ConverterDialog.InType.MULTI_FILE:
+            with wx.FileDialog(self, "Select Input Files", style=wx.FD_MULTIPLE, wildcard="*.ttl") as fileDialog:
+                res = fileDialog.ShowModal()
+                if res == wx.ID_OK:
+                    paths = fileDialog.GetPaths()
+                    self.__in_files = paths
+                    if len(self.__in_files) < 6:
+                        label.SetLabel("\n".join(self.__in_files))
+                    else:
+                        label.SetLabel(f"Selected files: {len(self.__in_files)}")
+        elif mode == ConverterDialog.InType.DIRECTORY:
+            with wx.DirDialog(self, "Select Input File") as dirDialog:
+                res = dirDialog.ShowModal()
+                if res == wx.ID_OK:
+                    path = dirDialog.GetPath()
+                    files = glob(f'{path}/*.ttl')
+                    self.__in_files = files
+                    if len(self.__in_files) < 6:
+                        label.SetLabel("\n".join(self.__in_files))
+                    else:
+                        label.SetLabel(f"Selected files: {len(self.__in_files)}")
```

Comment:
Probably this could be simplified, however I'd actually rather leave it like that because I found that simplifying things relating to `wxPython` is often a lot harder than it should be... and the present solution works at least.